src/data_hub/_api.py

```diff
@@ -111,6 +124,13 @@ def dynamic_function(*, client=None, **kwargs):
             else:
                 raise error
 
+        if isinstance(response, _response.APIResponse):
```

Comment:
Would `if debug:` be simpler?

The meaning of this if condition seems hard to understand. I understand the condition is when response has the type APIResponse. What type APIResponse means vs other possible types could be clearer. It seems like the condition for this is similar to `debug=True` and that this is related to `client = client.with_raw_response`.

Reply:
This makes sense. I'll change.

Reply:
fixed